src/entropy-keys.ts

```diff
@@ -37,6 +42,40 @@ export async function getPublicEntropyKey(): Promise<string> {
   return bytesToAddress(secp256k1.getPublicKey(privateKey));
 }
 
+// This is used to derive an encryption key from the entropy, to avoid key reuse.
+const staticSalt = 'metamask:snaps:encryption';
+
+/**
+ * Retrieve the secret encryption key for this snap.
+ * The key is derived from the entropy key and a static salt as sha256(entropy | staticSalt).
+ * @returns Secret Key Bytes.
+ */
+async function getEncryptionSecretKey(): Promise<Uint8Array> {
+  const privateEntropy = await getPrivateEntropyKey();
+  return sha256(concatBytes([privateEntropy, utf8ToBytes(staticSalt)]));
```

Comment:
Does EIP-1024 describe that the entropy should be concatenated with the salt and hashed? Otherwise you could use the salt property of `snap_getEntropy` instead.

Reply:
ERC1024 doesn't specify how the encryption key-pairs are generated from a given entropy, but rather which curve should be used (x25519), the message formats and algorithms.

I deliberately added a salt here to avoid key reuse.
The raw entropy is being used as a private key for signing but since the encryption key will be used for ECDH it's best to avoid reuse.

Reply:
Since the entropy provided to the snap is not easily replicable to some other context, I think it's less important here to use it directly as a secret key and more important to adhere to best practices, like avoiding key reuse.

If we ever needed to document the whole key generation process for this we would include these 2 steps:
1. SRP -> snap entropy (I don't know where to find this info yet)
2. snap entropy -> encryption secret (`sha256(entropy | 'metamask:snaps:encryption')`)

Reply:
Sorry, maybe I'm not understanding it properly, but how does it avoid key-reuse when using a static salt? It would be the same as calling `snap_getEntropy` with `staticSalt` as salt, as that generates a completely different key too.

Reply:
> SRP -> snap entropy (I don't know where to find this info yet)

SIP-6: https://metamask.github.io/SIPs/SIPS/sip-6

Reply:
> Sorry, maybe I'm not understanding it properly, but how does it avoid key-reuse when using a static salt? It would be the same as calling `snap_getEntropy` with `staticSalt` as salt, as that generates a completely different key too.

What I meant was that from the same entropy we need to derive 2 secret keys. One for encryption and one for signing. The point of avoiding key reuse is so that by some means the encryption key gets compromised, you don't automatically leak the signing key. This should be vice-versa, but sadly we are already using the unsalted entropy directly as the signing key.

You are, of course, right when stating that we can provide the salt directly to the `snap_getEntropy` call. I wasn't aware it accepts a salt parameter, so thank you for insisting on this!!!

Reply:
I updated the call to snap_getEntropy to achieve proper separation of concerns.
Thanks again for spotting this @Mrtenz !